components/EditTaskButton.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {useState} from "react";
+import { View, Button, Modal, StyleSheet, ScrollView, TextInput, Text, Pressable } from "react-native";
+import TimeCalender from "./TimeCalender";
+import ImagePickerSelector from "./ImagePickerSelector";
+
+type EditTaskButtonProps = {
+    label: string,
+};
+
+type SubmitProps = {
+    title: string,
+    startTime: Date,
+    endTime: Date,
+}
+
+//Button to make a Modal for editing a task pop up
+const EditTaskButton: React.FC<EditTaskButtonProps> = (props) => {
+    const [showModal, setShowModal] = useState(false);
+    const [submitProps, setSubmitProps] = useState<SubmitProps>({
+        title: props.label,
+        startTime: new Date(),
+        endTime: new Date(),
+    });
+    const handleTimeChange = (start: Date, end: Date) => {
+        setSubmitProps({
+            title: submitProps.title,
+            startTime: start,
+            endTime: end,
+        });
+    };
+
+    const handleSubmit = () => {
```

Comment:
I'm assuming that this is temporary

Reply:
Yes, when the endpoint is established, it will be used as an update activity